src/scripts/main.js

```diff
@@ -1,3 +1,418 @@
+Array.prototype.random = function() {
+  return this[Math.floor((Math.random() * this.length))];
+};
+
+const game = document.querySelector('.container');
+const startButton = game.querySelector('.button');
+const messages = game.querySelector('.message-container');
+const gameField = game.querySelectorAll('.field-cell');
+const startMessage = messages.querySelector('.message-start');
+const loseMessage = messages.querySelector('.message-lose');
+const winMessage = messages.querySelector('.message-win');
+const score = game.querySelector('.game-score');
+const fields = {};
+const colorClasses = ['field-cell--2', 'field-cell--4',
+  'field-cell--8', 'field-cell--16', 'field-cell--32',
+  'field-cell--64', 'field-cell--128', 'field-cell--256',
+  'field-cell--512', 'field-cell--1024', 'field-cell--2048'];
+const directions = {
+  ArrowUp: 'ArrowUp',
+  ArrowDown: 'ArrowDown',
+  ArrowLeft: 'ArrowLeft',
+  ArrowRight: 'ArrowRight',
+};
+const finalResult = 2048;
+
+startButton.addEventListener('click', () => {
+  if (startButton.classList.contains('start')) {
+    startButton.classList.remove('start');
+    startButton.classList.add('restart');
+    startButton.textContent = 'Restart';
+    startMessage.classList.add('hidden');
+
+    [...gameField].forEach((field, i) => {
+      fields[i] = field.innerText;
+    });
+  } else if (startButton.classList.contains('restart')) {
+    gameField.forEach(field => {
+      field.innerText = '';
+      field.classList.remove(...colorClasses);
+    });
+
+    for (let i = 0; i < 16; i++) {
+      fields[i] = '';
+    }
+    score.innerHTML = 0;
+    loseMessage.classList.add('hidden');
+    winMessage.classList.add('hidden');
+  }
+});
+
+document.addEventListener('keydown', event => {
+  if (!startButton.classList.contains('start')
+    && loseMessage.classList.contains('hidden')
+    && winMessage.classList.contains('hidden')) {
+    switch (event.key) {
+      case directions.ArrowUp:
+        upArrow(fields);
+        rematch(fields);
+        break;
+      case directions.ArrowDown:
+        downArrow(fields);
+        rematch(fields);
+        break;
+      case directions.ArrowLeft:
+        leftArrow(fields);
+        rematch(fields);
+        break;
+      case directions.ArrowRight:
+        rightArrow(fields);
+        rematch(fields);
+        break;
+      default:
+        return;
+    }
+
+    let freeFields = Object
+      .keys(Object
+        .fromEntries(Object
+          .entries(fields)
+          .filter(x => x[1] === '')));
+
+    const randomField = freeFields.random();
+    const addedNumber = Math.random() >= 0.1
+      ? 2
+      : 4;
+
+    fields[randomField] = addedNumber;
+    gameField[+randomField].textContent = addedNumber;
+
+    freeFields = Object
+      .keys(Object
+        .fromEntries(Object
+          .entries(fields)
+          .filter(x => x[1] === '')));
+
+    paint(gameField);
+
+    score.innerHTML = Object.values(fields).reduce((a, b) => +a + +b);
```

Comment:
The score should be increased based on the sum of merged cells, not the sum of all cells. Adjust the logic to calculate score correctly.